doc/spec.adoc

```diff
@@ -109,15 +156,60 @@ If channel does not directly define a stream, required channels will be searched
 
 If multiple channels are defined, the latest version from any channel that defines the stream (directly or through required channels) is used.
 
-If no stream that matches the artifact have been found, an error is returned to the caller.
+If no stream that matches the artifact have been found, version is resolved using fallback strategy defined in `resolves-if-no-stream` for the channel.
+An error is returned to the caller if
+* the fallback strategy is `none`
+* the fallback strategy is `latest`, `maven-latest` or `maven-release` but underlying Maven repository contains no metadata for artifacts with required `groupId` and `artifactId`
 
 If the stream defines a `version`, the artifact will be resolved based on this version. If that version of the artifact can not be pulled
 from the Maven repositories, an error is returned to the caller.
 If the stream defines a `versionPattern`, the version will be determined by querying the version of the artifacts from the
 Maven repositories and use the latest version that matches the pattern. If no version matches the pattern, an error is returned to the caller.
 
+#### Maven repository proxies
+
+A channel defines repositories using their `id` and a default `url`. When creating the channel from definition the provisioning tool can replace the provided `url` with URL of a proxy server or an alternative repository. The `id` of the repository must not be changed.
+
+### Blocking artifact versions
+When using an open channel, a situation may arise where certain artifacts are promoted to the channel and later discovered to be invalid. As it’s impossible to remove artifacts already deployed into a repository, those versions have to blocklisted.
+The blocklist is a separate YAML artifact included in the channel itself. When channel resolves the blacklist artifact, it will always resolve the latest available version. Any changes to the blocked artifact versions can be done independently of channel changes.
+#### Format
+The blocklist file contains a list of stream GAVs with multiple versions:
+```
+---
+exclusions:
+- groupId: io.undertow
+artifactId: undertow-core
+versions: [2.2.18.Final, 2.2.17.Final]
+…
+```
+The artifactId can use a wildcard syntax to refer to all the artifacts with the same groupId
+```
+---
+exclusions:
+- groupId: io.undertow
+artifactId: “*”
+versions: [2.2.18.Final, 2.2.17.Final]
+```
+
+#### Resolving channel blocklist
+
+The blocklist is resolved when a channel is initialized. If no blocklist artifact can be resolved with supplied GAVs, the channel treats it as an empty blocklist.
+A blocklist applies only to the channel that defined it, not its required channels.
+
+#### Resolving artifact version
+
+During artifact version resolution, a stream matching artifact GA is located in the channel. If the stream uses concrete versions, that version of the artifact is resolved and returned to the user.
```

Comment:
in that case, the blocklist is also checked too right?

Reply:
No it's not. 

The problem with blocklist and streams with concrete versions is that there is no version to fallback on. In "open" channel, we can return second-latest version if the latest one is blocked, in concrete channel, blocking the specified version would mean that this artifact cannot be resolved at all.

In channels with concrete versions excluding an artifact would mean publishing a new version of the manifest